scalding-core/src/main/scala/com/twitter/scalding/reducer_estimation/Common.scala

```diff
@@ -181,6 +187,13 @@ object ReducerEstimatorStepStrategy extends FlowStepStrategy[JobConf] {
       // if still None, make it '-1' to make it simpler to log
       val numReducers = combinedEstimator.estimateReducers(info)
 
+      numReducers.foreach { n =>
+        val configuredMax = conf.getInt(EstimatorConfig.maxEstimatedReducersKey, EstimatorConfig.defaultMaxEstimatedReducers)
+        if (n > configuredMax) {
+          throw new IllegalArgumentException(s"Reducer estimator estimated $n reducers, which is more than the configured maximum of $configuredMax")
```

Comment:
Yeah, although it is not ideal behavior from the standpoint of a user. So if there is a bug, it ends up paging the user, not the developers of the reducer estimator. In this case, failing fast would be a bad user experience IMO.